src/app/admin/entity.go

```diff
@@ -0,0 +1,37 @@
+package admin
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type Domain struct {
+	Name string
+	NIP  string
+	User UserReference
+}
+
+type UserReference struct {
+	ID        uuid.UUID
+	Email     string
+	Password  string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type Services interface {
+	GetAllAdmins() (admins []Domain, err error)
+	GetAdminByID(id string) (admin Domain, err error)
+	CreateAdmin(admin Domain) (id string, err error)
+	AmendAdminByID(id string, admin Domain) (err error)
+	RemoveAdminByID(id string) (err error)
+}
+
+type Repositories interface {
+	SelectAllData() (data []Domain, err error)
+	SelectDataByID(id string) (data Domain, err error)

```

Comment:
Prefer to return using a pointer
example:
SelectDataByID(id string) (data *Domain, err error)

Reply:
![image](https://user-images.githubusercontent.com/59078748/173177355-8a6d1d1d-6214-4e9a-857a-8a0c6bbc64bc.png)

Is this acceptable?

Reply:
Because golang doesn't support return pointer of return value of called function directly.

Reply:
it will get nil pointer dereference

Reply:
I suggest creating a new variable and returning it. ex:
varName := MapToDomain(record)
return &varName, nil

Reply:
Done at https://github.com/ALTERA-CAPSTON-41/backend/pull/23/commits/b115e6ae6343514826981bf681852cb3287387df, mocks https://github.com/ALTERA-CAPSTON-41/backend/pull/23/commits/3170333f50c5412f53bc613f61736fd0eb1d2418 and its test case https://github.com/ALTERA-CAPSTON-41/backend/pull/23/commits/e1e0280ca18a0bc352cda2e2ff38a4963ef6699f.